crates/chia-datalayer/src/lib.rs

```diff
@@ -1,42 +1,155 @@
-use std::collections::HashMap;
+// use std::collections::HashMap;
 
-type Index = usize;
-type Key = Vec<u8>;
+type TreeIndex = u32;
```

Comment:
Instead of a type alias, you could make a newtype like this:
```rs
#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
struct TreeIndex(32);
```
This has the benefit that it will not be compatible with plain `u32` integers by default for type safety, and the actual index isn't exposed publicly. You can of course make the index `pub` if you need, or add an `fn index(&self) -> u32` method.

Docs on the newtype idiom:
https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Reply:
I did want the type safety aspect but wasn't convinced by the newtype wrapper approach quite yet.  I do expect to try it out at some point and see how it feels for this.

Reply:
It's a really common pattern. For example, a `NodePtr` is a just a `u32`:
https://github.com/Chia-Network/clvm_rs/blob/main/src/allocator.rs#L14

The `.0` and whatnot is annoying at first but I got used to it.